src/EdgeDB.Net.Driver/Attributes/EdgeDBTypeConverterAttribute.cs

```diff
@@ -0,0 +1,30 @@
+using EdgeDB.TypeConverters;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace EdgeDB
+{
+    [AttributeUsage(AttributeTargets.Property, AllowMultiple = false)]
+    public class EdgeDBTypeConverterAttribute : Attribute
+    {
+        internal IEdgeDBTypeConverter Converter;
```

Comment:
Can this be public?

Reply:
The interface `IEdgeDBTypeConverter` is internal, what would the usecase for making this public be?

Reply:
Could that be public too? Would be great when attempting to use reflection

Reply:
The worry I have is people will attempt to implement the interface and not the class. The interface is an abstracted form of the class without type strictness on whats serialized and deserialized, its only used at the convert step to avoid having to use reflection-based generic def's.

Reply:
Gotcha. There's definitely a way around this, however, I cannot think of an intuitive solution at the current moment.

Reply:
Reflection isn't stopping you getting the non-public property 😃

Reply:
Good point

Reply:
Not my area to comment, but if there's enough concern, you may warrant dundering the interface so people know it's a no-touchey.